src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/Common/CommonTypes.Windows.cs

```diff
@@ -58,6 +58,10 @@ internal unsafe struct ComObjectFactory
         [ModuleInitializer]
         internal static unsafe void RegisterInProcCOMServer()
         {
+            // Make sure we are initialized for COM before registering
```

Comment:
How do you know that it is ok to initialize the COM apartment on the thread that this happens to run on?

This fix looks fragile. I think a better fix is to stop using ModuleInitializer for the initialization, and instead trigger the initialization e.g. from the static constructor of the tests types that depend on this.

Reply:
Ah, I took another look through all this. We're always initializing threads in coreclr by default to MTA (and document it as such). For NativeAOT, we initialize new threads to MTA, but only initialize the main thread if it explicitly has `STAThread` or `MTAThread`. So I think we actually want to make NativeAOT match the coreclr/documented behaviour of initializing the main thread to MTA by default.